src/ValueParsers/StringParser.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace ValueParsers;
+
+use InvalidArgumentException;
+use DataValues\StringValue;
+use ValueParsers\Normalizers\NullStringNormalizer;
+use ValueParsers\Normalizers\StringNormalizer;
+
+/**
+ * Implementation of the ValueParser interface for StringValues.
+ *
+ * @since 0.2.4
+ *
+ * @licence GNU GPL v2+
+ * @author Daniel Kinzler
+ */
+class StringParser implements ValueParser {
+
+	/**
+	 * @var StringNormalizer
+	 */
+	private $normalizer;
+
+	/**
+	 * @param StringNormalizer $normalizer
+	 */
+	public function __construct( StringNormalizer $normalizer = null ) {
+		if ( $normalizer === null ) {
```

Comment:
Having a single assignment would be better than having two. No need for this much state